extract-wos-article.py

```diff
@@ -23,7 +30,54 @@
     "Current Publisher": "publisher",
 }
 
-YEAR_REGEX = re.compile(".*([0-9]{4})")
+REGEX = {
+    "year": re.compile(".*([0-9]{4})"),
+    "wos_id_number": re.compile("([0-9]{15})"),
+}
+CITING_SUMMARY_URL_PREFIX = "https://www.webofscience.com/wos/woscc/citing-summary/"
+
+
+async def extract_citing_summary(page: Page, article_id: str):
+    url = CITING_SUMMARY_URL_PREFIX + article_id
+    await page.goto(url, wait_until="networkidle")
+
+    curr_page = await page.query_selector(CITING_SELECTORS["current_page"])
+    curr_page_text = await curr_page.input_value()
+    curr_page_num = int(curr_page_text.strip())
+
+    final_page = await page.query_selector(CITING_SELECTORS["final_page"])
+    final_page_text = await final_page.inner_html()
+    final_page_num = int(final_page_text.strip())
+
+    articles = []
+    while True:
+        curr_articles_num = -1
+        curr_articles = []
+
+        while len(curr_articles) > curr_articles_num:
+            curr_articles_num = len(curr_articles)
+            await page.keyboard.press("PageDown")
+            curr_articles = await page.query_selector_all(
+                CITING_SELECTORS["citations_articles"]
+            )
+            time.sleep(0.1)
```

Comment:
Just so that I make sure I am getting this right, you are pressing `PageDown` here so that new articles show up? In that case, could we perhaps try to wait for some text to appear (maybe via  [`page.locator`](https://playwright.dev/python/docs/api/class-locator#locator-wait-for)) rather than sleeping here?

Reply:
@mrshu any idea what should i bind the `page.locator` on? see the video below. it shows how the articles are rendered onto the page one by one, only if you manually scroll down through them.

https://user-images.githubusercontent.com/19410533/147783710-52f49804-68c4-4791-93dc-547429a334d6.mov

Reply:
@JMatej well, I am pretty sure this could be figured out somehow (i.e. by locating the `Links` button or some other part of text) but I am also positive it is not worth it. Feel free to keep it as it is, at least for now :slightly_smiling_face: